app/knight.py

```diff
@@ -0,0 +1,126 @@
+        self.name = knight_dict["name"]
+
+        self.armour = knight_dict["armour"]
+        self.weapon = knight_dict["weapon"]
+        self.potion = knight_dict["potion"]
+
+        self.hp = knight_dict["hp"]
+        self.power = knight_dict["power"] + self.weapon["power"]
+        self.protection = 0
+
+        for arm in self.armour:
+            self.protection += arm["protection"]
+
+        if self.potion is not None:
+            if "power" in self.potion["effect"]:
+                self.power += self.potion["effect"]["power"]
+
+            if "protection" in self.potion["effect"]:
+                self.protection += self.potion["effect"]["protection"]
+
+            if "hp" in self.potion["effect"]:
+                self.hp += self.potion["effect"]["hp"]
+
+    def duel(self, enemy: Knight) -> None:
+        self.hp -= enemy.power - self.protection
+        enemy.hp -= self.power - enemy.protection
+
+        if self.hp <= 0:
+            self.hp = 0
+
+        if enemy.hp <= 0:
+            enemy.hp = 0
+
+
+KNIGHTS = {
+    "lancelot": {
+        "name": "Lancelot",
+        "power": 35,
+        "hp": 100,
+        "armour": [],
+        "weapon": {
+            "name": "Metal Sword",
+            "power": 50,
+        },
+        "potion": None,
+    },
+    "arthur": {
+        "name": "Artur",
+        "power": 45,
+        "hp": 75,
+        "armour": [
+            {
+                "part": "helmet",
+                "protection": 15,
+            },
+            {
+                "part": "breastplate",
+                "protection": 20,
+            },
+            {
+                "part": "boots",
+                "protection": 10,
+            }
+        ],
+        "weapon": {
+            "name": "Two-handed Sword",
+            "power": 55,
+        },
+        "potion": None,
+    },
+    "mordred": {
+        "name": "Mordred",
+        "power": 30,
+        "hp": 90,
+        "armour": [
+            {
+                "part": "breastplate",
+                "protection": 15,
+            },
+            {
+                "part": "boots",
+                "protection": 10,
+            }
+        ],
+        "weapon": {
+            "name": "Poisoned Sword",
+            "power": 60,
+        },
+        "potion": {
+            "name": "Berserk",
+            "effect": {
+                "power": +15,
+                "hp": -5,
+                "protection": +10,
+            }
+        }
+    },
+    "red_knight": {
+        "name": "Red Knight",
+        "power": 40,
+        "hp": 70,
+        "armour": [
+            {
+                "part": "breastplate",
+                "protection": 25,
+            }
+        ],
+        "weapon": {
+            "name": "Sword",
+            "power": 45
+        },
+        "potion": {
+            "name": "Blessing",
+            "effect": {
+                "hp": +10,
+                "power": +5,
+            }
+        }
+    }
+}
```

Comment:
Why did you put it here?)

Reply:
I tried to make `main.py` smaller, but it's useless. And also I forgot that when importing `battle()`, this dictionary will also be imported as redundant